client/components/drop-zone/index.jsx

```diff
@@ -144,6 +153,25 @@ export const DropZone = React.createClass( {
 			// from tracked nodes since another "real" event will be triggered.
 			this.dragEnterNodes = without( this.dragEnterNodes, window );
 		}
+
+		this.toggleDropZoneReduxState( this.state.isDraggingOverDocument || this.state.isDraggingOverElement );
+	},
+
+	toggleDropZoneReduxState( isVisible ) {
+		/**
+		 * The DropZone component is included in components/libraries which are not yet reduxified,
+		 * for which `connect` ( from `react-redux` ) complains that it can't find `store`
+		 * in the current context.
+		 *
+		 * For places where the reduxification has happened, this will work.
+		 */
+		if ( this.context.store ) {
+			if ( isVisible ) {
+				this.context.store.dispatch( showDropZone() );
+			} else {
+				this.context.store.dispatch( hideDropZone() );
+			}
+		}
```

Comment:
Thanks for the input, Bernie. Per a couple of Slack chats, we're going to address it ASAP in the following PR. This one can then serve as the base for both that fix and for the proper DnD functionality being pushed.